### ISSUE TITLE
Screen readers can view / focus content behind some modals / dialogs

### ISSUE THREAD

**Issue by microbit-robert** [microsoft/pxt-microbit#6740]
**Describe the bug**
Screen readers can view / focus content behind modals that use [this Modal component](https://github.com/microsoft/pxt/blob/master/react-common/components/controls/Modal.tsx). 

**To Reproduce**
1. Go to https://makecode.microbit.org/#editor
2. Click on the 'share project' button
3. With VoiceOver enabled, use the VoiceOver key and arrow keys to focus / read content behind the dialog, or use the VoiceOver rotor (VO key + u) to navigate by landmark to do this, for example

**Expected behavior**
The background content should be aria-hidden or inert so that screen readers cannot interact with it. This is the case for [this Modal component](https://github.com/microsoft/pxt/blob/master/webapp/src/sui.tsx#L1161).

**Screenshots**
N/A

**micro:bit version (please complete the following information):**
Not hardware related

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Chrome Version 145.0.7632.160 (Official Build) (arm64)

**Additional context**
The working Modal implementation uses a portal to render the dialog in `document.body` and the main content in `#content` (the parent of `#root`) is made `aria-hidden`.

The Modal implementation that exhibits the issue described uses a portal to render the dialog inside an optional parent element, this falls back to `#root` and then `document.body`. In the share project dialog case, the parent element is `#root`, so `#content` cannot be made aria-hidden or inert as it includes the dialog. A general fix would be to render the dialogs in `document.body` and make `#content` inert, but moving these dialogs without additional checks could have unintended consequences (e.g., broken CSS). An alternative could be adding aria-hidden / inert on all sibling elements to the dialog.

Finally, the `common-focus-trap` container element itself with `tabindex="-1"` can also and unexpectedly be focused by the screen reader.
